File: cli/src/commands/stats.ts
import { requireApiKey } from '../config.js'
import { api } from '../api.js'
import { accent, accentBold, dim } from '../theme.js'

interface Stats {
  totalPosts: number
  totalReplies: number
  totalUsers: number
  totalVotes: number
  totalBoards: number
  postsToday: number
}

export async function statsCommand(): Promise<void> {
  const key = requireApiKey()
  const res = await api<Stats>('/stats', key)
  const stats = res.data!

  console.log()
  console.log(accentBold('Arkora Stats'))
  console.log()
  console.log(`  Posts:     ${accent(stats.totalPosts.toLocaleString())}`)
  console.log(`  Replies:   ${accent(stats.totalReplies.toLocaleString())}`)
  console.log(`  Users:     ${accent(stats.totalUsers.toLocaleString())}`)
  console.log(`  Votes:     ${accent(stats.totalVotes.toLocaleString())}`)
  console.log(`  Boards:    ${accent(stats.totalBoards.toLocaleString())}`)
  console.log(`  Today:     ${accent(stats.postsToday.toLocaleString())} posts`)
  console.log()
}

File: cli-rust/src/commands/stats.rs
use anyhow::Result;
use serde::Deserialize;

use crate::api::ArkoraApi;
use crate::config;
use crate::theme;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct Stats {
    total_posts: i64,
    total_polls: i64,
    total_verified_humans: i64,
    total_poll_votes: i64,
}

pub async fn run(color: (u8, u8, u8)) -> Result<()> {
    let cfg = config::load();
    let key = config::require_key(&cfg);
    let api = ArkoraApi::new(&config::api_url(&cfg), &key);

    let resp = api.get::<Stats>("/stats").await?;
    let s = resp.data.ok_or_else(|| anyhow::anyhow!("No data in response"))?;

    println!();
    println!("{}", theme::accent_bold("Arkora Stats", color));
    println!();
    println!("  Posts:      {}", theme::accent(&s.total_posts.to_string(), color));
    println!("  Polls:      {}", theme::accent(&s.total_polls.to_string(), color));
    println!("  Humans:     {}", theme::accent(&s.total_verified_humans.to_string(), color));
    println!("  Poll votes: {}", theme::accent(&s.total_poll_votes.to_string(), color));
    println!();

    Ok(())
}
